tests/test_x.py

```diff
@@ -184,10 +184,10 @@ def test_set_dense_x_view_from_sparse():
     assert_equal(orig.X[30:], x[30:])  # change propagates through
 
 
-def test_warn_on_non_csr_csc_matrix():
-    X = sparse.eye(100)
```

Comment:
which `format` does `eye` return and why are you switching to `coo`?

Reply:
It was returning `DIA` before but I figured I'd just make it explicit here that it's not `cs{r,c}` (exactly because of your first question).